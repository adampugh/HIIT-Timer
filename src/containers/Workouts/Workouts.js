import React, { Component } from "react";
import { connect } from "react-redux";


import Navbar from "../../components/UI/navbarLoggedIn";
import WorkoutsBanner from "./workoutsbanner";
import WorkoutsGrid from "./workoutsGrid";
import Footer from "../../components/UI/footer";
import * as actions from "../../store/actions/actions";

class Workouts extends Component {
    state = {
        createModalOpen: false,
        deleteModalOpen: false
    }

    componentWillMount() {
        this.props.startFetchWorkouts();
    }

    openCreateModal = () => {
        this.setState({
            createModalOpen: true
        });
    }

    closeCreateModal = () => {
        this.setState({
            createModalOpen: false
        });
    }

    openDeleteModal = () => {
        this.setState({
            deleteModalOpen: true
        });
    }

    closeDeleteModal = () => {
        this.setState({
            deleteModalOpen: false
        });
    }

    render() { 
        return (
            <div>
                <Navbar />
                <WorkoutsBanner />
                <WorkoutsGrid 
                    workouts={this.props.workouts} 
                    createModalIsOpen={this.state.createModalOpen}
                    handleOpenCreateModal={this.openCreateModal}
                    handleCloseCreateModal={this.closeCreateModal} 
                    deleteModalIsOpen={this.state.deleteModalOpen}
                    handleOpenDeleteModal={this.openDeleteModal}
                    handleCloseDeleteModal={this.closeDeleteModal} />
                <Footer />
            </div> 
        );
    }
}

const mapStateToProps = state => {
    return {
        workouts: state.reducer.workouts
    }
}

const mapDispatchToProps = dispatch => {
    return {
        startFetchWorkouts: () => dispatch(actions.startFetchWorkouts())
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(Workouts);