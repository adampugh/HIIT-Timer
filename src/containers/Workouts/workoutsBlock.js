import React from "react";

const workoutsBlock = (props) => (
    <div className="workoutsBlock">
        <h1>{props.title}</h1>
        <h3>{props.totalTime} Minutes</h3>
        <i className="far fa-play-circle"></i>
    </div>
);

export default workoutsBlock;