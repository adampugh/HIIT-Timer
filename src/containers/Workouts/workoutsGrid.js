import React from "react";
import { Link } from "react-router-dom";

import WorkoutsBlock from "./workoutsBlock";
import CreateModal from "../../components/UI/createModal";

const workoutsGrid = (props) => (
    <div>
        <CreateModal 
            isOpen={props.isOpen}
            handleCloseModal={props.handleCloseModal} />
        <div className="workouts__wrapper">
            <div className="container">
                <div className="workouts__grid">
                    {props.workouts.map(workout =>
                        <div key={workout.title}> 
                            <Link to={`/workouts/${workout.title}/start`}>
                                <WorkoutsBlock 
                                    title={workout.title}
                                    totalTime={workout.totalTime} />
                            </Link>
                            <div className="workoutsBlock__edit">
                                <Link to={`/workouts/${workout.title}`}>
                                    <h3><i className="far fa-edit"></i> Edit</h3>
                                </Link>
                                <h3><i className="far fa-trash-alt"></i> Delete</h3>
                            </div>
                        </div>
                    )}
                    <div onClick={props.handleOpenModal} className="workouts__addBlock">
                        <h1>Create a workout</h1>
                        <i className="fas fa-plus-circle"></i>
                    </div>
                </div>
            </div>
        </div>
    </div>
);

export default workoutsGrid;