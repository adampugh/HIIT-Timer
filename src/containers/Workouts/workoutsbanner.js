import React from "react";

const workoutsBanner = () => (
    <div className="bannerText">
        <div className="bannerText__container">
            <h2>
                Add a new workout and start today! Click the + button below to add a new workout.
            </h2>
        </div>
    </div>
);

export default workoutsBanner;