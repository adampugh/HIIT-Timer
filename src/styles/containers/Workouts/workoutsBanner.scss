.workoutsBanner__image {
    height: 100px;
}