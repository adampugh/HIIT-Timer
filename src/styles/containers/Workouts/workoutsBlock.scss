.workoutsBlock {
    margin: 0 auto;
    text-align: center;
    justify-self: center;
    align-self: center;
    border: 5px solid white;
    background: $purple-gradient;
    border-radius: 20px;
    color: white;
    height: 200px;
    overflow-wrap: break-word;
    padding: 40px;
    max-width: 200px;
    cursor: pointer;
    @include boxshadow;
    i, svg {
        font-size: 40px;
    }
    h1 {
        height: 100px;
        margin: 0;
    }
}

.workoutsBlock__edit {
    text-align: center;
    color: grey;
    margin: 0 auto 80px auto;
    width: 250px;
    h3 {
        cursor: pointer;
    }
}

@media only screen and (min-width: $desktop-breakpoint) {
    .workoutsBlock {
        -webkit-box-shadow: none;
        -moz-box-shadow: none;
        box-shadow: none;
        &:hover {
            @include boxshadow;
        }
    }
}