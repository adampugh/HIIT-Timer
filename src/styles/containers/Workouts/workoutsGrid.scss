.workouts__grid {
    display: grid;
    grid-template-columns: 1fr;
    padding-bottom: 100px;
}

.workouts__wrapper {
    background: $light-grey;
}

.workouts__addBlock {
    text-align: center;
    justify-self: center;
    align-self: center;
    border: 5px dashed grey;
    border-radius: 20px;
    margin-bottom: 80px;
    padding: 40px;
    width: 200px;
    height: 200px;
    cursor: pointer;
    h1 {
        height: 168px;
        margin: 0;
    }
    svg, i {
        font-size: 40px;
    }
    
}

@media only screen and (min-width: $desktop-breakpoint) {
    .workouts__grid {
        grid-template-columns: 1fr 1fr 1fr;
    }

    .workouts__addBlock {
        &:hover {
            @include boxshadow;
        }
    }
}