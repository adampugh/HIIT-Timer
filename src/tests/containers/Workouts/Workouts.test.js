import React from "react";
import { shallow } from "enzyme"; 

import { Workouts } from "../../../containers/Workouts/Workouts";
import workoutState from "../../fixtures/workouts";

test("should render Workouts correctly", () => {
    const wrapper = shallow(<Workouts 
        workouts={workoutState.workouts}
        startFetchWorkouts={() => {}} />);
    expect(wrapper).toMatchSnapshot();
});