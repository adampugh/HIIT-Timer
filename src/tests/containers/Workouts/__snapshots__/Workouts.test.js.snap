// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Workouts correctly 1`] = `
<div>
  <Connect(Component) />
  <workoutsBanner />
  <workoutsGrid
    createModalIsOpen={false}
    deleteModalIsOpen={false}
    handleCloseCreateModal={[Function]}
    handleCloseDeleteModal={[Function]}
    handleOpenCreateModal={[Function]}
    handleOpenDeleteModal={[Function]}
    workouts={
      Array [
        Object {
          "exercises": Array [
            Object {
              "color": "#40b3e1",
              "id": "L2453",
              "name": "squats",
              "time": 60,
            },
            Object {
              "color": "#40b3e1",
              "id": "L2435y",
              "name": "pressups",
              "time": 60,
            },
          ],
          "id": "L6ERTEQ",
          "title": "Leg Workout",
          "totalTime": 60,
        },
        Object {
          "exercises": Array [
            Object {
              "color": "#40b3e1",
              "id": "L24df3",
              "name": "box jumps",
              "time": 180,
            },
            Object {
              "color": "#40b3e1",
              "id": "L2fdgy",
              "name": "sit ups",
              "time": 160,
            },
          ],
          "id": "L6sdg45",
          "title": "Shoulder Workout",
          "totalTime": 30,
        },
      ]
    }
  />
  <footer />
</div>
`;
